Samples/iOS-Swift6.xcconfig

```diff
@@ -0,0 +1,27 @@
+#include "Shared/Config/Architectures.xcconfig"
+#include "Shared/Config/BuildOptions.xcconfig"
+#include "Shared/Config/Deployment.xcconfig"
+#include "Shared/Config/Linking.xcconfig"
+#include "Shared/Config/Localization.xcconfig"
+#include "Shared/Config/Packaging.xcconfig"
+#include "Shared/Config/SearchPaths.xcconfig"
+#include "Shared/Config/Signing.xcconfig"
+#include "Shared/Config/Versioning.xcconfig"
+#include "Shared/Config/CodeGeneration.xcconfig"
+#include "Shared/Config/ClangLanguage.xcconfig"
+#include "Shared/Config/ClangCppLanguage.xcconfig"
+#include "Shared/Config/ClangModules.xcconfig"
+#include "Shared/Config/ClangObjCLanguage.xcconfig"
+#include "Shared/Config/ClangPreprocessing.xcconfig"
+#include "Shared/Config/ClangWarnings.xcconfig"
+#include "Shared/Config/ClangWarningsCpp.xcconfig"
+#include "Shared/Config/ClangWarningsObjC.xcconfig"
+#include "Shared/Config/AssetCatalog.xcconfig"
+#include "Shared/Config/ClangAnalyzer.xcconfig"
+#include "Shared/Config/Swift.xcconfig"
+#include "Shared/Config/Metal.xcconfig"
+
+PRODUCT_BUNDLE_IDENTIFIER = io.sentry.sample.iOS-Swift6
+INFOPLIST_FILE = iOS-Swift6/Info.plist
+SUPPORTED_PLATFORMS = iphoneos iphonesimulator
+SWIFT_VERSION = 6
```

Comment:
`m`: Can we somehow make `iOS-Swift6` use the exact same configuration as `iOS-Swift` and only set the `SWIFT_VERSION` to `6`? I think this sample is used not as a separate platform like macOS/tvOS etc.

Reply:
we can, i considered rolling up the xcconfigs to a common config that's included in each target config and then overriding only what's necessary, but wanted to do it in a later refactor so that change is isolated and to keep these PRs as compartmentalized as possible

Reply:
noted this in https://github.com/getsentry/sentry-cocoa/issues/5165#issuecomment-2852482982